docs/config.md

```diff
@@ -318,6 +319,100 @@ location /blog/ {
 
 Default: `[]`
 
+- `NoDNSLink`  
+  A boolean to configure whether DNSLink lookup for value in `Host`
+  HTTP header should be performed.  If DNSLink is present, content path stored
+  in the DNS TXT record is mounted at `/` and respective payload is returned to
+  the client.
+
+Default: `false`
+
+### Gateway.PublicGateways
+
+`PublicGateways` is a dictionary for defining gateway behavior on specific fully qualified domain names (FQDN).  
+Options for changing behavior per hostname:
+
+- `Paths`  
+  Array of paths that should be mounted at the root of domain name.  
+  Example: `["/ipfs", "/ipns", "/api"]`  
+  - **Note:** when both `Paths` and DNSLink are mounted, `Paths` takes priority
+
+- `UseSubdomains`
+  A boolean to configure whether the gateway provides [Origin isolation](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy)
+  between content roots.  
+  Default: `false`
+  - `true` – mount [subdomain gateway](#https://docs-beta.ipfs.io/how-to/address-ipfs-on-web/#subdomain-gateway) at `http://{cid}.ipfs.{hostname}`
+    - requires respective `Paths` to be set
+    - requests for `http://{hostname}/ipfs/{cid}` will return redirect to `http://{cid}.ipfs.{hostname}`
+  - `false` – mount [path gateway](https://docs-beta.ipfs.io/how-to/address-ipfs-on-web/#path-gateway) at `http://{hostname}/ipfs/{cid}`
+    <!-- **(not implemented yet)** due to the lack of Origin isolation, cookies and storage on `Paths` will be disabled by [Clear-Site-Data](https://github.com/ipfs/in-web-browsers/issues/157) header -->
```

Comment:
Can we do this now? This sounds like it's pretty easy to do.

Reply:
Sure. I will add a flag to disable this type of hardening if someone really needs to do it:

> Blocking cookies is a nuclear option that may break some deployments behind reverse proxies that pass all headers. We would need `NoWebSecurity` flag (`false` by default) in case someone really wants to disable things like `Clear-Site-Data` in contexts that provide no origin isolation. 


I'll go with `NoWebSecurity` (`false` by default) and make it work the same as `NoDNSLink` (unless you suggest something else).

Reply:
SGTM.

Reply:
Note to self: just for sake of merging this sooner than later, we agreed to descope `Clear-Site-Data` from this PR  and land it separately :)